core/ui/PageTemplate/modals.tid

```diff
@@ -0,0 +1,87 @@
+title: $:/core/ui/PageTemplate/modals
+tags: $:/tags/PageTemplate
+list-after: 
+created: 20230214123242192
+modified: 20230215183342466
+
+\whitespace trim
+
+\define close-current-modal()
+<$action-listops $tiddler="$:/state/modal" $field="text" $filter="[enlist:raw{$:/state/modal}butlast[]]" />
+<!-- Delete paramObject tiddler -->
+<$action-deletetiddler $tiddler={{{ [[$:/state/modal/paramObject/]addsuffix<modalCounter>] }}} />
+\end
+
+\define tm-close-handler()
+<!-- Close the modal unless a different tiddler was given in the message parameter -->
+<$let currentModal={{{ [enlist:raw{$:/state/modal}last[]] }}} >
+<$list filter="[<event-param>!is[blank]!match<currentModal>]" emptyMessage=<<close-current-modal>> >
+<$action-sendmessage $message="tm-close-tiddler" />
+</$list>
+</$let>
+\end
+
+\define tm-modal-handler()
+<$action-listops $tiddler="$:/state/modal" $field="text" $filter="[enlist:raw{$:/state/modal}] :all[<event-param>]" />
+<!-- Store the paramObject hashmap to set variables inside the modal -->
+<$let targetCount={{{ [<modalCounter>!is[blank]else[0]add[1]] }}}>
+<$action-setmultiplefields $tiddler={{{ [[$:/state/modal/paramObject/]addsuffix<targetCount>] }}}
+		  				   $indexes="[enlist<list-event-paramObject>]"
+						   $values="[enlist<list-event-paramObject>addprefix[event-paramObject-]getvariable[]]" />
+</$let>
+\end
+
+<style>
+.tc-modal-footer {
+	display: flex;
+}
+.tc-modal-footer > span:last-of-type {
+	flex-grow: 1;
+}
+
+</style>
+
+<!-- Do we have a list of modals to display in the state tiddler? -->
+<$reveal stateTitle="$:/state/modal" stateField="text" type="nomatch" text="" tag="div" class="tc-modal-wrapper">
+<$navigator story=<<tv-story-list>>
+            history=<<tv-history-list>>
+			openLinkFromOutsideRiver={{$:/config/Navigation/openLinkFromOutsideRiver}} >
+	<!-- Show all modal tiddlers in the order they are listed in the state tiddler -->
+	<$list filter="[enlist:raw{$:/state/modal}]" variable="currentModal" counter="modalCounter" >
+	<!-- We need special handling for tm-close-tiddler and tm-modal from inside the modal (the latter to be handled by the core eventually) -->
+	<$messagecatcher $tm-close-tiddler=<<tm-close-handler>> $tm-modal=<<tm-modal-handler>> >
+	<!-- Retrieve the paramObject hashmap in case a modal was called with it -->
+	<$setmultiplevariables $names="[[$:/state/modal/paramObject/]addsuffix<modalCounter>indexes[]]" $values="[[$:/state/modal/paramObject/]addsuffix<modalCounter>indexes[]getindex[]]">
+	<!-- Implements mask-closable (only checks for "yes") -->
+	<$eventcatcher $click={{{ [<currentModal>get[mask-closable]match[yes]then<close-current-modal>] }}} selector=".tc-modal-backdrop" matchSelector=".tc-modal-backdrop" tag="div">
+	<!-- Display modal content -->
+	<div class="tc-modal-backdrop">
```

Comment:
This structure is wrong. .. `tc-modal-backdrop` must not cover the `tc-modal`. As the name says it is a "backdrop". The modals need to be on top of it. 

That means the `matchSelector` of the eventcatcher needs to be changed too. I'll have a closer look. 

If `tc-modal-backdrop` covers the whole thing and it's style is set to `opacity: .7;` as it has to be, it will also make the modal semi-transparent.

Reply:
You're right. `tc-modal-backdrop` and `tc-modal` must be siblings. I'll change that. I think `matchSelector` might not be needed in this case at all...

Like I said, I didn't touch the styles yet. The opacity is set via JavaScript in modal.js, to fade out the background in an animation.